internal/util/util.go

```diff
@@ -135,17 +135,30 @@ func GetVolumeSnapshotClassForStorageClass(provisioner string, snapshotClient sn
 }
 
 // GetVolumeSnapshotContentForVolumeSnapshot returns the volumesnapshotcontent object associated with the volumesnapshot
-func GetVolumeSnapshotContentForVolumeSnapshot(volSnap *snapshotv1beta1api.VolumeSnapshot, snapshotClient snapshotter.SnapshotV1beta1Interface, log logrus.FieldLogger) (*snapshotv1beta1api.VolumeSnapshotContent, error) {
+func GetVolumeSnapshotContentForVolumeSnapshot(volSnap *snapshotv1beta1api.VolumeSnapshot, snapshotClient snapshotter.SnapshotV1beta1Interface, log logrus.FieldLogger, wait bool) (*snapshotv1beta1api.VolumeSnapshotContent, error) {
```

Comment:
I'll look at this in some more detail in the morning, but I think we may be able to simplify by using some of the helpers from `k8s.io/apimachinery/pkg/util/wait`.

Reply:
I'll also look for something we can use from there.

Reply:
`wait.PollImmediate` would probably be a good fit